### ISSUE TITLE
Add rule for not deleted reference after foreach

### ISSUE THREAD

**Issue by fre5h** [phpmd/phpmd#63]
``` php
foreach ($something as &$item) {
    $item['new_key'] = 'new value';
}

unset($item); // Reference should be deleted here
// If reference is not deleted then phpmd should give warning

foreach ($somethingElse as &$item) {
    $item['new_key'] = 'new value';
}
// Or other manipulation with $item

```

I think this rule will be useful.
